client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/gateway/FakeRemoteGateway.kt

```diff
@@ -279,6 +298,34 @@ class FakeRemoteGateway : IRemoteGateway {
         return getRestaurants().filter { it.name.startsWith(restaurantName, true) }
     }
 
+    override suspend fun filterRestaurants(rating: Double, priceLevel: Int): List<Restaurant> {
+        return getRestaurants().filter {
+            it.priceLevel == priceLevel &&
+                    when {
+                        rating.rem(1) > 0.89 || rating.rem(1) == 0.0 || rating.rem(1) > 0.5
+                        -> it.rating in floor(rating) - 0.1..0.49 + floor(rating)
+
+                        else -> it.rating in 0.5 + floor(rating)..0.89 + floor(rating)
+                    }
+        }
+    }
+
+    override suspend fun searchFilterRestaurants(
+        restaurantName: String,
+        rating: Double,
+        priceLevel: Int
+    ): List<Restaurant> {
+        return searchRestaurantsByRestaurantName(restaurantName).filter {
+            it.priceLevel == priceLevel &&
+                    when {
+                        rating.rem(1) > 0.89 || rating.rem(1) == 0.0 || rating.rem(1) > 0.5
+                        -> it.rating in floor(rating) - 0.1..0.49 + floor(rating)
+
```

Comment:
i see you have you repeated this logic , you can put it in a function will be more clean ♥